bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/SomfyTahomaHandlerFactory.java

```diff
@@ -184,6 +185,8 @@ protected ThingHandler createHandler(Thing thing) {
             return new SomfyTahomaMyfoxAlarmHandler(thing);
         } else if (thingTypeUID.equals(THING_TYPE_THERMOSTAT)) {
             return new SomfyTahomaThermostatHandler(thing);
+        } else if (thingTypeUID.equals(THING_TYPE_WATERHEATINGSYSTEM)) {
+            return new SomfyTahomaWaterHeatingSystemHandler(thing);
```

Comment:
Please use deeper detection using getWidget(), I guess there might be more water heating systems and this makes the autodetection more accurate. See the CLASS_CAMERA detection as an example...